### PR TITLE
pull rsvp question data from meetup

### LINKED ISSUE #2: pull answer data from rsvp endpoint
## Problem

When people rsvp for events they complete answers to questions, and we are not capturing this.

## Solution

* Capture RSVP data from meetup api, https://www.meetup.com/meetup_api/docs/:urlname/events/:event_id/rsvps/#list.

### Requirement 
* do this for each meetup event id we have
*  capture answers
    * meetup api gives us of answers, store each answer as a separate row
* also capture the flowing info for each answer
  * meetup question
  * meetup question id
  * meetup member id 
  * meetup group urlname (whatever is the quickest reference)

### Gotcha
* answers and questions may not always exist
* specify `fields=answers` as a url param so you can get the questions back

# Blocking
~~~by #4 and #3~~~

### PR DESCRIPTION
**Pull request by isometimescode**
closes #2 

## Description:
- Using the Meetup Event RSVP endpoint, grabs question and answer data and inserts it into the database

## QA:
- [ ] Run tests: `heroku run bundle exec rspec `
- [x] Run rake task, which uses the seed data to grab RSVP questions: `heroku run bundle exec rake data_import:rsvps`
- [x] Examine RSVP data in the console: `heroku run pry`

## Notes:

On staging I guess we'll need to run migrations:
- [ ] `heroku run bundle exec rake db:migrate`

@WomenWhoCode/meet-maynard-reviewers

### CODE REVIEW

File: lib/tasks/data_import.rake [hunk 1]
```diff
@@ -24,4 +24,14 @@ namespace :data_import do
       Event.retrieve_events(group_stat)
     end
   end
+
+  desc "pull rsvp data. Usage: rake data_import:rsvps['Women-Who-Code-Silicon-Valley']"
+  task :rsvps, [:urlname] do |t, args|
+    scope = args[:urlname].present? ? Event.where(group_urlname: args[:urlname]) : Event.all
```
Comment (zassmin) [inline lib/tasks/data_import.rake]: each time we turn this it will pull rsvp data for all events in our db if we don't pass in the urlname?

Comment (isometimescode) [inline lib/tasks/data_import.rake]: Correct; urlname = events for only that one group, no urlname = all events that are in the db

Comment (zassmin) [inline lib/tasks/data_import.rake]: Can we make it so we only grab rsvp data for events that don't already have it stored? This data is in the past, so it doesn't need to get updated.

Comment (isometimescode) [inline lib/tasks/data_import.rake]: Ah ok, I wasn't sure if the data changed at all. I can make it grab events that don't already have RSVP question/answers stored.

**Comment by isometimescode** [comment]
@zassmin added a scope to Events to skip records that have RSVP questions already. Ready for re-review!

File: app/models/rsvp_question.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+          meetup_data = m.get_next_page(Date.today)
+        end
+
+      rescue => e
+        Bugsnag.notify(e, event_id: event.id)
+    end
+
+    def create_from_answers(data)
+      return nil unless data["answers"] && data["answers"].any?
```
Comment (zassmin) [inline app/models/rsvp_question.rb]: nitpick: returning nil if blank? will have the same effect

File: app/models/rsvp_question.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+      fields = {
+        event_id: data["event"]["id"],
+        group_urlname: data["group"]["urlname"],
+        member_id: data["member"]["id"],
+      }
+
+      data["answers"].each do |ans|
```
Comment (zassmin) [inline app/models/rsvp_question.rb]: Assuming I ran the task when this PR was checking if answers exist... it looks like we might have to add an additional somewhere as I'm seeking blank answers returned. 🤔

image

maybe another check needs to be added here?

File: app/models/rsvp_question.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  class << self
+    def retrieve_answers(event)
+        m = Meetup::Api.new(
+              data_type: [event.group_urlname, "events", event.event_id, "rsvps"],
+              options: {fields: "answers"}
+            )
+
+        meetup_data = m.get_response
+        while meetup_data && meetup_data.any?
```
Comment (zassmin) [inline app/models/rsvp_question.rb]: nitpick: while not blank? will have the same effect

File: app/models/rsvp_question.rb (hunk 3 above)
Comment (isometimescode) [inline app/models/rsvp_question.rb]: @zassmin added a second check to skip answers that are blank

**Pull request merged**